File: 3151 special-array-i/solution.go
func isArraySpecial(nums []int) bool {
    e:=(nums[0] & 1) ==0
    for i:=1;i<len(nums);i++ {
        c:=(nums[i] & 1) == 0
        if c==e {
            return false
        }
        e = c
    }
    return true
}

File: 3151 special-array-i/solution.java
class Solution {
    public boolean isArraySpecial(int[] nums) {
        boolean isEven = (nums[0] & 1) == 0;
        for(int i=1;i<nums.length;i++) {
            boolean currEven = (nums[i] & 1) == 0;
            if(isEven == currEven) return false;
            isEven = currEven;
        }
        return true;
    }
}
